common/helpers.go

```diff
@@ -46,6 +46,26 @@ func initEnviron() {
 	}
 }
 
+// CleanURL is similar to path.Clean() but to work on URLs
+func CleanURL(url string) (string, error) {
+	elements := strings.Split(url, "/")
+	retval := ""
+	if len(elements) < 3 {
+		return "", errors.New("Invalid URL")
+	}
+	retval = elements[0] + "//" + elements[2] 
```

Comment:
Yeah it doesn't clean everything everythign; but it is enough to deal with an issue where ServeHTTP in golang behaves weirdly in // in path. Should we really thorough clean it?